arithmetic_analysis/newton_raphson_method.py

```diff
@@ -4,31 +4,35 @@
 # quickly find a good approximation for the root of a real-valued function
 from sympy import diff
 from decimal import Decimal
+from math import sin,log,exp
+
+
+PRECISION = 10 ** -10
 
 
 def NewtonRaphson(func, a):
     """ Finds root from the point 'a' onwards by Newton-Raphson method """
```

Comment:
How about some doctests...

"""
>>> NewtonRaphson("sin(x)", 2))
0
>>> NewtonRaphson("x**2 - 5*x +2", 0.4))
0
>>> NewtonRaphson("x**2 - 5", 0.1))
0
>>> NewtonRaphson("log(x)- 1", 2))
0
"""
     # Exponential Roots
     print("exp(x) - 1 = 0", NewtonRaphson("exp(x) - 1", 0))
     print("The root of exp(x) - 1 = 0 is", NewtonRaphson("exp(x) - 1", 0))
"""